lib/connection.js

```diff
@@ -963,19 +980,31 @@ Connection.prototype._close = function(force, callback) {
         if (err) {
           return callback(err);
         }
+        if (closeAndClean && _this.base.connections.indexOf(conn) != -1) {
+          console.log("BASE CONNECTIONS COUNT BEFORE 2---------> ", _this.base.connections.length)
+          _this.base.connections.splice(_this.base.connections.indexOf(conn), 1);
+          console.log("BASE CONNECTIONS COUNT AFTER 2---------> ", _this.base.connections.length)
+
+        }
         _this.onClose(force);
         callback(null);
       });
 
       break;
     case STATES.connecting:
       this.once('open', function() {
-        _this.close(callback);
+        console.log("======== connecting ==========")
```

Comment:
We need to remove this one as well.

Reply:
Yes, I am testing out these changes, will remove them once testing is done before I mark this ready to merge.